t/39-dewebsockify.t

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/perl
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Mojo::Base -strict, -signatures;
+use Test::Most;
+use Test::MockModule;
+use Test::Warnings;
+use Mojo::Log;
+use Mojo::IOLoop::Server;
+
+BEGIN {
+    require OpenQA::Isotovideo::Dewebsockify;
+}
+
+my $args_common = {
+    websocketurl => 'ws://example.com',
+    listenport => 5900,
+    cookie => undef,
+    loglevel => 'info',
+};
+
+my ($accept_callback, @log_messages);
+our ($ws_on_text, $ws_on_binary, $ws_on_finish);
+
+my $mock_log = Test::MockModule->new('Mojo::Log');
+$mock_log->redefine(info => sub ($self, $msg) { push @log_messages, $msg });
+$mock_log->redefine(error => sub ($self, $msg) { push @log_messages, $msg });
+$mock_log->redefine(trace => sub ($self, $msg) { push @log_messages, $msg });
+
+my $mock_server = Test::MockModule->new('Mojo::IOLoop::Server');
+$mock_server->redefine(
+    new => sub { bless {}, 'Mojo::IOLoop::Server' },
+    on => sub ($self, $event, $cb) {
+        $accept_callback = $cb if $event eq 'accept';
+    },
+    listen => sub { },
+    start => sub { },
+);
+
+my $mock_stream = Test::MockModule->new('Mojo::IOLoop::Stream');
+$mock_stream->redefine(
+    new => sub { bless {}, 'Mojo::IOLoop::Stream' },
+    start => sub { },
+    on => sub { },    # selectively overridden in subtests
+    write => sub { },
+    close_gracefully => sub { },
+);
+
+my $mock_ua = Test::MockModule->new('Mojo::UserAgent');
+$mock_ua->mock(new => sub { bless {}, 'Mojo::UserAgent' });
+$mock_ua->mock(start => sub ($ua, $tx, $cb) {
+        $cb->($ua, $tx);
+});
+
+{
+    package MockTxGeneric;
```

Comment:
Can you try the suggestion from the meeting? It's a bit shorter
```suggestion
package MockTxGeneric {
```

Reply:
I don't see you tried it. Have you tried it and it didn't work?

Reply:
Have you seen [my comment](https://github.com/os-autoinst/os-autoinst/pull/2598#issuecomment-2586787234) below ? Sorry, i should've commented here or @ mention you